torch/distributed/__init__.py

```diff
@@ -18,6 +29,22 @@ def is_available():
 if is_available() and not torch._C._c10d_init():
     raise RuntimeError("Failed to initialize torch.distributed")
 
+def get_debug_mode():
+    """
+    Reads the environment variable ``TORCH_DISTRIBUTED_DEBUG`` and returns a
+    ``DistributedDebugMode`` corresponding to the appropriate debug level. The
+    env var ``TORCH_DISTRIBUTED_DEBUG`` must be set to one of "OFF", "INFO", or
+    "DETAIL". Default setting is INFO.
```

Comment:
Shouldn't default be OFF?

Reply:
We can probably do off for now while this is under development, although I think the proposal was to go with INFO by default and verify that it doesn't have a significant perf hit. DETAIL would have things such as the ProcessGroupWrapper that can affect performance. But during development we can stick with OFF, and measure the perf impact of DETAIL when switching over